DeeployTest/testUtils/typeMapping.py

```diff
@@ -68,25 +68,26 @@ def inferInputType(_input: np.ndarray,
         print(f"Warning: Empty input array for type inference for {_input}!")
         return [(defaultType, defaultOffset)]
 
-    if not isInteger(_input):
-        raise Exception("Deeploy currently only handles integer types!")
-
     if signProp is None:
         signProp = False
 
     signedPlatformTypes = [_type for _type in IntegerDataTypes if _type.typeMin < 0]
 
     matchingTypes = []
 
-    if signProp and isUnsigned(_input):
+    if signProp and isUnsigned(_input) and isInteger(_input):
         for _type in sorted(signedPlatformTypes, key = lambda x: x.typeWidth):
             signPropOffset = (2**(_type.typeWidth - 1))
             if _type.checkPromotion(_input - signPropOffset):
                 matchingTypes.append(offsetType(PointerClass(_type), signPropOffset))
-    else:
+    elif isInteger(_input):
         for _type in sorted(IntegerDataTypes, key = lambda x: x.typeWidth):
             if _type.checkPromotion(_input):
                 matchingTypes.append(offsetType(PointerClass(_type), 0))
+    else:
+        for _type in sorted(FloatDataTypes, key = lambda x: x.typeWidth):
+            if _type.checkPromotion(_input):
+                matchingTypes.append(offsetType(PointerClass(_type), 0))
```

Comment:
NOTE: I think this is okay for now (3 distinctions are fine), but there is implicit knowledge encoded in the order of the checks (i.e. first unsigned, signed and then float). It might be good to extract that implicit knowledge into an ordered list.

Reply:
I left the code as is, but added your comment as a `FIXME` above it.